torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -4413,35 +4413,50 @@ Tensor lu_solve_jvp(
 }
 
 Tensor lu_unpack_backward(
-  const variable_list& grads,
-  const Tensor& LU_data,
-  bool unpack_data
+  const Tensor& L_grad,
+  const Tensor& U_grad,
+  const int64_t m,
+  const int64_t n
 ) {
-  auto L_grad = grads[1];
-  auto U_grad = grads[2];
-
-  auto m = LU_data.size(-2);
-  auto n = LU_data.size(-1);
-  auto k = std::min(m, n);
-
-  TORCH_CHECK(unpack_data, "lu_unpack_backward: cannot compute gradients unless unpack_data=True");
+  if (!L_grad.defined() && !U_grad.defined()) {
```

Comment:
logic and is necessary here and cannot be replaced with or?

Reply:
What do you mean here? If both gradients are zero, then the input gradient is zero, but we cannot say much more otherwise.